src/components/modal/Modal.tsx

```diff
@@ -0,0 +1,60 @@
+import { FunctionComponent, useRef } from 'react'
+import { IModalProps } from './IModalProps'
+import DefaultCloseIcon from './DefaultCloseIcon'
+import './Modal.css'
+
+const Modal : FunctionComponent<IModalProps> 
+= ({
+    setShowModal, 
+    showModal,
+    CloseIcon,
+    title,
+    Body,
+    footer,
+    rounded
+}) => {
+    const modalRef = useRef<HTMLDivElement>(null)
+
+    const handleOnClose = () => {
+        setShowModal(false)
+    }
+
+    const handleOnClick = (e : any) => {
+        if (modalRef.current === e.target)
+            handleOnClose()
+    }
+
+    return (
+        <>
+        {
+            (showModal) 
+            ? (
+                <div className='modal__background' ref={modalRef} onClick={handleOnClick}>
+                    <div className={`modal__container ${rounded? 'modal__rounded' : ''}`}>
+                        <div className='modal__header'>
+                            <p>{ title }</p>
+                            <button onClick={handleOnClose} className='modal__header__close'>
```

Comment:
Here what we want is to receive the whole actionable element via prop, not only the icon. Suppose I want to use a link instead of a button to close the modal, but I don't want to be in charge of handling the event. Can you think a way to achieve this using React APIs?

Reply:
Before I started implementing the exercise, I took a look into your repos. I saw this function from React Api, which I had never used before. I think that I can use it to achieve this
<img width="593" alt="Captura de pantalla 2022-03-21 a las 13 04 55" src="https://user-images.githubusercontent.com/51116828/159248815-2716c543-5671-4ce5-94e3-3750e433e534.png">